internal/animation/animation.go

```diff
@@ -13,11 +14,28 @@ type anim struct {
 	reverse     bool
 	start       time.Time
 	total       int64
+
+	mu      sync.RWMutex
+	stopped bool
 }
 
 func newAnim(a *fyne.Animation) *anim {
+	// TODO should we add a nil check here? (to avoid panic)
```

Comment:
Always good to be safe. Normally we would only do nil checks for public APIs, but maybe this one is called blindly?

Reply:
Yes, you are right, good catch! That situation is impossible to happen. I will remove that TODO.

Reply:
done